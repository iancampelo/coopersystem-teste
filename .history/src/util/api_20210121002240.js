import axios from 'axios'

const getInvestimentos = axios.create({
    baseURL: 'https://api.openweathermap.org/data/2.5',
    timeout: 30000,
    headers: {
        'Content-Type': 'application/json',
    },
})

const apiLocationInstance = axios.create({
    baseURL: 'http://api.positionstack.com/v1',
    timeout: 30000,
    headers: {
        'Content-Type': 'application/json',
    },
})

const appId = 'a83d800af84b3adb088ce541ea2920d3'
const access_key = '39e9e11c1ebdae135d30ab6fd7fe8f49'

const getOneCall = (lat, lon) => apiWeatherInstance.get(`/onecall?lat=${lat}&lon=${lon}&units=metric&appid=${appId}`)

const getPastDays = (lat, lon, time) => apiWeatherInstance.get(`/onecall/timemachine?lat=${lat}&lon=${lon}&dt=${time}&appid=${appId}`)

const getLatLonFromLocation = (query) => apiLocationInstance.get(`/forward?access_key=${access_key}&query=${query}&limit=1&output=json`)

export default {
    getOneCall,
    getPastDays,
    getLatLonFromLocation,
}
