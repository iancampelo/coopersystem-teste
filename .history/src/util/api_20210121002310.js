import axios from 'axios'

const getInvestimentos = axios.create({
    baseURL: 'http://www.mocky.io/v2/5e76797e2f0000f057986099',
    timeout: 30000,
    headers: {
        'Content-Type': 'application/json',
    },
})

const getOneCall = (lat, lon) => apiWeatherInstance.get(`/onecall?lat=${lat}&lon=${lon}&units=metric&appid=${appId}`)

const getPastDays = (lat, lon, time) => apiWeatherInstance.get(`/onecall/timemachine?lat=${lat}&lon=${lon}&dt=${time}&appid=${appId}`)

const getLatLonFromLocation = (query) => apiLocationInstance.get(`/forward?access_key=${access_key}&query=${query}&limit=1&output=json`)

export default {
    getOneCall,
    getPastDays,
    getLatLonFromLocation,
}
