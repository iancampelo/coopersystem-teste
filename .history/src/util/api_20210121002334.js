import axios from 'axios'

const apiInvestimentos = axios.create({
    baseURL: 'http://www.mocky.io/v2',
    timeout: 30000,
    headers: {
        'Content-Type': 'application/json',
    },
})

const getInvestimentos = (lat, lon) => apiWeatherInstance.get(`/onecall?lat=${lat}&lon=${lon}&units=metric&appid=${appId}`)

const getPastDays = (lat, lon, time) => apiWeatherInstance.get(`/onecall/timemachine?lat=${lat}&lon=${lon}&dt=${time}&appid=${appId}`)

const getLatLonFromLocation = (query) => apiLocationInstance.get(`/forward?access_key=${access_key}&query=${query}&limit=1&output=json`)

export default {
    getOneCall,
    getPastDays,
    getLatLonFromLocation,
}
