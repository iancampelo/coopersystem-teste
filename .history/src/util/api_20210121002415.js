import axios from 'axios'

const apiInvestimentos = axios.create({
    baseURL: 'http://www.mocky.io/v2',
    timeout: 30000,
    headers: {
        'Content-Type': 'application/json',
    },
})

const getInvestimentos = () => apiInvestimentos.get(`/5e76797e2f0000f057986099`)

export default {
  getInvestimentos,
}
