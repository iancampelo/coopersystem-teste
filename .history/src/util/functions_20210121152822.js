export const per = (temp) => {
  return `${Math.round(temp - 273.15)}Â°`
}
