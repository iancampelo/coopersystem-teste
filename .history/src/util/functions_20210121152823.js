export const percentual = (temp) => {
  return `${Math.round(temp - 273.15)}Â°`
}
