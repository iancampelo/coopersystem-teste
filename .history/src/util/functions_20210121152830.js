export const percentualToValue = (temp) => {
  return `${Math.round(temp - 273.15)}Â°`
}
