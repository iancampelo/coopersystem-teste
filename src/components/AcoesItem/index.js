import React, { useState } from 'react'
import {
    StyleSheet,
    View,
    Text,
} from 'react-native'
import CurrencyInput from 'react-currency-input'
import { normalizeMoneyString } from '../../util/functions'

const AcoesItem = (props) => {
  const [errorMessage, setErrorMessage] = useState('');
  const {acao, limit, inputValue} = props

  const handleChange = (event, maskedValue, floatValue) => {
    props.changeHandler(event, maskedValue, floatValue)
    if (Number(floatValue) > limit) {
      setErrorMessage(`Valor não pode ser maior que ${normalizeMoneyString(limit)}`)
    } else {
      setErrorMessage('');
    }
  }

  return (
    <View style={styles.item}>
      <View style={styles.card}>
        <Text style={styles.subtitle}>Ação</Text>
        <Text>{acao.nome}</Text>
      </View>
      <View style={styles.card}>
        <Text style={styles.subtitle}>Saldo acumulado</Text>
        <Text>{normalizeMoneyString(limit)}</Text>
      </View>
      <CurrencyInput
        style={{height: 20}}
        placeholder="Valor a resgatar"
        onChangeEvent={handleChange}
        value={inputValue}
        id={acao.nome}
        allowEmpty={true}
        prefix="R$ "
        decimalSeparator=","
        thousandSeparator="."
        inputType="tel"
      />
      <Text>{errorMessage}</Text>
    </View>
  )
}

const styles = StyleSheet.create({
  item: {
    padding: 15,
    height: 100,
    backgroundColor: "white",
    marginBottom: 10
  },
  card: {
    flex: 1,
    justifyContent: "space-between",
    flexDirection: "row",
    marginBottom: 10
  },
  subtitle: {
    fontWeight: "bold",
  },
});

export default AcoesItem
