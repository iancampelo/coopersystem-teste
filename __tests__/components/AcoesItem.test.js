import 'react-native'
import React from 'react'
import AcoesItem from '../../src/components/AcoesItem'
import { percentualToValor } from '../../src/util/functions'

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer'
jest.useFakeTimers()

jest.mock('react-currency-input', () => props => <input type="text" {...{ ...props }} />)

const mockAcoes = {
  nome: 'INVESTIMENTO I',
  objetivo: 'Minha aposentadoria',
  saldoTotalDisponivel: 39321.29,
  indicadorCarencia: 'N',
  acoes: [
    {
      id: '1',
      nome: 'BBAS3',
      percentual: 28.1,
    },
    {
      id: '2',
      nome: 'VALE3',
      percentual: 20.71,
    },
    {
      id: '3',
      nome: 'PETR4',
      percentual: 21.63,
    },
    {
      id: '4',
      nome: 'CMIG3',
      percentual: 12.41,
    },
    {
      id: '5',
      nome: 'OIBR3',
      percentual: 17.15,
    },
  ],
};

it('renders correctly AcoesItem', () => {
  const tree = renderer.create(
    <AcoesItem
      acao={mockAcoes.acoes[0]}
      limit={percentualToValor(mockAcoes.saldoTotalDisponivel, mockAcoes.acoes[1].percentual)}
      changeHandler={jest.fn()}
      inputValue={jest.fn()}
    />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})
