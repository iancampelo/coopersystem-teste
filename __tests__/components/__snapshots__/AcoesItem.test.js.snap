// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly AcoesItem 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "height": 100,
      "marginBottom": 10,
      "padding": 15,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginBottom": 10,
      }
    }
  >
    <Text
      style={
        Object {
          "fontWeight": "bold",
        }
      }
    >
      Ação
    </Text>
    <Text>
      BBAS3
    </Text>
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginBottom": 10,
      }
    }
  >
    <Text
      style={
        Object {
          "fontWeight": "bold",
        }
      }
    >
      Saldo acumulado
    </Text>
    <Text>
      R$8,143.44
    </Text>
  </View>
  <input
    allowEmpty={true}
    decimalSeparator=","
    id="BBAS3"
    inputType="tel"
    onChangeEvent={[Function]}
    placeholder="Valor a resgatar"
    prefix="R$ "
    style={
      Object {
        "height": 20,
      }
    }
    thousandSeparator="."
    type="text"
    value={[MockFunction]}
  />
  <Text>
    
  </Text>
</View>
`;
